MultiLoggerSingleton2/Logger.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace MultiLoggerSingleton2
+{
+    public class Logger: ILogger
+    {
+        public static int m_count = 0;
+        private static readonly object lockObject = new object();
+        private static Dictionary<string, Logger> m_object = new Dictionary<string, Logger>();
+
+        // private constructor
+        private Logger()
+        {
+            m_count++;
```

Comment:
Why we need this m_count

Reply:
just an extra check for Singleton. 
I used this variable to check how many times our constructor is getting called.